src/components/select/utils/get-next-child-by-key-code.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+
+export function getNextChildByKeyCode(keyCode, children, previousSelectedValue) {
+  const childrenList = React.Children.toArray(children);
+  const currentSelectedIndex = childrenList.findIndex((option) => {
+    return option.props.value === previousSelectedValue;
+  });
+  const lastIndex = childrenList.length - 1;
+
+  return childrenList[getNextIndexByKey(keyCode, currentSelectedIndex, lastIndex)];
+}
+
+export function getNextIndexByKey(keyCode, currentIndex, lastIndex) {
+  const isNoOptionSelected = currentIndex === -1;
+  let newIndex = currentIndex;
+
+  if (keyCode === 36) { // home
```

Comment:
You used the `Event` class in the other files, e.g. `Events.isHomeKey` `Events.isEndKey` `Events.isDownKey` and `Events.isUpKey`. Should we use them here also?

Reply:
Event class works on events, I'd have to pass down whole event and then import that again there.

I've removed the dependency on the `Event` util as it's using deprecated `which` event property.

Reply:
keyCode is also deprecated
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

We should probably use `e.key === 'Home`. Some of the methods in the Events util already do this.

Reply:
keyCode was the name of a variable, it's not related to the event. I have changed that variable name to key as it's no longer keyCode based.